### ISSUE TITLE
[typescript-angular] Disable tslint in generated files /* tslint:disable */

### ISSUE THREAD

**Issue by etrich** [swagger-api/swagger-codegen#6720]
##### Description

The generated TS code disables only specific tslint rules:

/* tslint:disable:no-unused-variable member-ordering */

Some other generated files do not disable any tslint rules.

The generated code is causing tslint issues when using with Codelyzer, such as:

- " should be '
- file should end with a newline
- trailing whitespace
etc.

##### Swagger-codegen version

2.3.0 - swagger-codegen-cli-2.3.0-20171004.083430-178

##### Steps to reproduce

- Install codelyzer
- Run `ng lint`

##### Suggest a fix/enhancement

Simple fix would be to disable all tslint rules in generated files:

/* tslint:disable  */

**Comment by macjohnny** [comment]
@etrich could you file a pull request?

**Comment by wing328** [comment]
I would suggest we fix those warnings/issues reported by tslint one by one instead of disabling tslint.

@etrich would you have time to help file PRs to make the TS Angular generator better? (this is not to say we push the task to you but I'm sure the community would appreciate your help to make the TS Angular generator better in terms of the code format)

**Comment by etrich** [comment]
> I would suggest we fix those warnings/issues reported by tslint one by one instead of disabling tslint.

Given the rules can be dynamically configured in tslint.json, it would require a generator that can be dynamically configured using these rules and able to produce code in accordance. I don't think anyone wants to go there, so we'd need to agree on a tslint ruleset, which could be the tslint.json produced by Angular CLI `ng new` (which uses codelyzer).

For people wanting to use "non standard" tslint rules, there would still be an opt-out in Angular CLI by adding lines in .angular-cli.json:

"lint": [{
      "project": "src/tsconfig.app.json",
      "exclude": ["**/node_modules/**", "**custom-path**"]
    },

I can probably find some time to help, what's release date for 2.3.0 ?

**Comment by wing328** [comment]
> I can probably find some time to help, what's release date for 2.3.0 ?

We hope to release it by the end of this year.

**Comment by zimme** [comment]
I'll just a note here about tslint's no-unused-variables being deprecated in favor of tsc's --noUnusedLocals and friends. Which might be something have in mind.

**Comment by JoshuaKGoldberg** [comment]
Filed a related issue for typescript-fetch at #10835, but reposting here:

* [TSLint is deprecated](https://github.com/palantir/tslint/issues/4534)
* For autogenerated codefiles, you probably should not be running a linter such as ESLint or TSLint on them anyway -- they're not a part of our source code

Would you accept a PR to remove references to TSLint?

**Comment by quezak** [comment]
@JoshuaKGoldberg good point! I have the generated files excluded in my `ignored` eslintrc section.

However, I actually do run `eslint --fix` from my scripts once after the code is generated, to get rid of unused imports etc, that could later cause a typescript compilation error. Not that this matters here, just wanted to provide a point of view :)
